### PR TITLE
KAFKA-10437: Update WordCount examples to use new PAPI

### PR DESCRIPTION
**Pull request by dengziming**
*More detailed description of your change*
It seems that #9396 leaves out a TODO, just fix it.

*Summary of testing strategy (including rationale)*
QA

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by dengziming** [comment]
Hello @vvcephei, PTAL.

**Comment by dengziming** [comment]
@mjsax @guozhangwang , how do you think about this approach? In fact, I'm not very confident about this change since it isn't very elegant, but this can remove the TODO, feel free to point my fault.

**Comment by mjsax** [comment]
@dengziming What TODO are you referring to? I was trying to find a comment on the old PR that describes this "TODO". Can you elaborate?

**Comment by dengziming** [comment]
Hello @mjsax , I mean the TDDO at 
https://github.com/apache/kafka/blob/6de37e536ac76ef13530d49dc7320110332cd1ee/streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java#L36

File: streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java [hunk 1]
```diff
@@ -45,11 +50,26 @@ public void test() {
...
-        transformer.init(context);
+        transformer.init(new org.apache.kafka.streams.processor.MockProcessorContext() {
+            @Override
+            public <S extends StateStore> S getStateStore(final String name) {
+                return context.getStateStore(name);
+            }
+
+            @Override
+            public <K, V> void forward(final K key, final V value) {
+                context.forward(new Record<>(key.toString(), value.toString(), 0L));
```
Comment (mjsax) [inline streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java]: Why `toString()` -- should we not assume that both `K` and `V` are of type `String`? I guess we cannot declare the input parameters as `String`, but wondering if we should rather cast here?

File: streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java [hunk 2]
```diff
@@ -45,11 +50,26 @@ public void test() {
...
+            }
+
+            @Override
+            public Cancellable schedule(final Duration interval, final PunctuationType type, final Punctuator callback) throws IllegalArgumentException {
```
Comment (mjsax) [inline streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java]: nit: remove `throws` declaration (not needed).

**Comment by mjsax** [comment]
Thanks for clarifying. Left a few minor comments. Overall LGTM.

File: streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java (hunk 1 above)
Comment (dengziming) [reply streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java]: Yes, both K and V are types of String here, we should use cast here.

File: streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java (hunk 2 above)
Comment (dengziming) [reply streams/examples/src/test/java/org/apache/kafka/streams/examples/wordcount/WordCountTransformerTest.java]: Done!

**Pull request merged**